File: aoc2024/java/src/test/java/com/peeekay/aoc2024/java/Day04Test.java
package com.peeekay.aoc2024.java;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class Day04Test {
    final Day04 day = new Day04(true);

    @Test
    void part1() {
        day.solve();
        Assertions.assertEquals(18, day.part1());
    }

    @Test
    void part2() {
        day.solve();
        Assertions.assertEquals(9, day.part2());
    }
}

File: aoc2024/kotlin/src/test/kotlin/Day04Test.kt
package com.peeekay.aoc2024.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 04")
internal class Day04Test {

    @Test
    fun `part1 test data solve gives correct answer`() {
        val day = Day04(isTest = true)
        val answer = day.partOne()

        assertEquals(18, answer)
    }

    @Test
    fun `part2 test data solve gives correct answer`() {
        val day = Day04(isTest = true)
        val answer = day.partTwo()

        assertEquals(0, answer)
    }

}
